 lib/metadata/raid_manip.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/metadata/raid_manip.c b/lib/metadata/raid_manip.c
index c71c084..9e26c0e 100644
--- a/lib/metadata/raid_manip.c
+++ b/lib/metadata/raid_manip.c
@@ -1627,6 +1627,8 @@ int lv_raid_replace(struct logical_volume *lv,
 		return 0;
 	}
 
+	usleep(500000);
+
 	if (!mirror_in_sync() && !_raid_in_sync(lv)) {
 		log_error("Unable to replace devices in %s/%s while it is"
 			  " not in-sync.", lv->vg->name, lv->name);
