 conf/example.conf.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/conf/example.conf.in b/conf/example.conf.in
index 8e2bf57..ef04048 100644
--- a/conf/example.conf.in
+++ b/conf/example.conf.in
@@ -74,7 +74,7 @@ devices {
     # preferred_names = [ ]
 
     # Try to avoid using undescriptive /dev/dm-N names, if present.
-    # preferred_names = [ "^/dev/mpath/", "^/dev/mapper/mpath", "^/dev/[hs]d" ]
+    preferred_names = [ "^/dev/mpath/", "^/dev/mapper/mpath", "^/dev/[hs]d" ]
 
     # In case no prefererred name matches or if preferred_names are not
     # defined at all, builtin rules are used to determine the preference.
