 lib/activate/activate.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/activate/activate.c b/lib/activate/activate.c
index 6823e25..b4e3a65 100644
--- a/lib/activate/activate.c
+++ b/lib/activate/activate.c
@@ -2061,12 +2061,12 @@ int monitor_dev_for_events(struct cmd_context *cmd, const struct logical_volume
 		} else
 			continue;
 
-		if (!locking_is_clustered() && !vg_write_lock_held() && lv_is_mirror(lv)) {
+		if (!cmd->is_clvmd && !vg_write_lock_held() && lv_is_mirror(lv)) {
 			/*
 			 * Commands vgchange and lvchange do use read-only lock when changing
 			 * monitoring (--monitor y|n). All other use cases hold 'write-lock'
 			 * so they skip this dm mirror table refreshing step.
-			 * Shortcut can't be applied with clustered locking.
+			 * Shortcut is also not applied with clvmd.
 			 */
 			if (!_lv_activate_lv(lv, &mirr_laopts)) {
 				stack;
